clang/lib/Sema/SemaConcept.cpp

```diff
@@ -636,6 +636,9 @@ bool Sema::SetupConstraintScope(
             ? FD->getInstantiatedFromMemberFunction()
             : FD->getInstantiatedFromDecl();
 
+    if (!InstantiatedFrom)
+      return true;
+
```

Comment:
This is unnecessary.
For both `TK_MemberSpecialization` and `TK_DependentNonTemplate`, in `FD->getTemplatedKind()`, we had ensured that `getMemberSpecializationInfo()` or `getInstantiatedFromDecl()` would always return non-null.

Reply:
Thank you @zyn0217 for the reviews and clarification. Closing this as Not a defect.